apps/web/pages/[[...root]].tsx

```diff
@@ -0,0 +1,74 @@
+import { useRouter } from "next/router";
+
+import { ComponentTree } from "../components";
+import { useWebEngine } from "../hooks";
+import { useEffect } from "react";
+
+const DEFAULT_COMPONENT = process.env.NEXT_PUBLIC_DEFAULT_ROOT_COMPONENT;
+
+export default function Root() {
+  const router = useRouter();
+  const { query } = router;
+
+  const isDebug = query.isDebug === "true";
+  const showMonitor = query.showMonitor === "true";
+  const rootComponentPath = Array.isArray(query.root)
+    ? query.root.join("/")
+    : null;
+
+  useEffect(() => {
+    if (router.isReady && !query.root && DEFAULT_COMPONENT) {
+      // change URL in place to accurately reflect default param values
+      router.push(`/${DEFAULT_COMPONENT}`, undefined, { shallow: true });
+    }
+  }, [router.isReady, query.root]);
+
+  return (
+    <div className="App">
+      {rootComponentPath && (
+        <WebEngineRoot
+          rootComponentPath={rootComponentPath}
+          isDebug={isDebug}
+          showMonitor={showMonitor}
+        />
+      )}
+    </div>
+  );
+}
+
+// conditionally rendered when rootComponentPath is defined so that
+// we do not have to account for invocations of hook before router
+// params are ready for consumption
```

Comment:
I am torn about this approach. Hooks can't be called conditionally so it feels cleaner in our application to move the `useWebEngine` call to a component which is conditionally rendered instead of building `useWebEngine` to handle `rootComponentPath` being `undefined` before the router query is ready to be consumed. In effect, this ends up building an expectation of a check always being made outside `useWebEngine` which is probably more error prone if that hook is ever used anywhere else

Technically this implementation more faithfully fulfills the `useWebEngine` props type that declares `rootComponentPath` as `string` (as opposed to `string?`) than the previous implementation where we were passing an empty string before the query was ready

Reply:
That's a good point on practical usage. The default behavior for empty paths would be to fail validation and surface in `errors`. Then the entire render becomes

```jsx
  return (
    <div className="App">
      {error && (
        <div className="error">{error}</div>
      )}
      {rootComponentPath && (
        <ComponentTree ... />
      )}
    </div>
  );
```